network_diagnostics.go

```diff
@@ -0,0 +1,35 @@
+package storageos
+
+import (
+	"encoding/json"
+	"net/http"
+	"path"
+
+	"github.com/storageos/go-api/types"
+)
+
+var (
+	// NetworkDiagnosticsAPIPrefix is a partial path to the HTTP endpoint for
+	// the node connectivity diagnostics report.
+	NetworkDiagnosticsAPIPrefix = "diagnostics/network"
+)
+
+// NetworkDiagnostics returns a collection of network connectivity reports.  If
+// a reference to a node is given, it will only check connectivity from that
+// node.  Otherwise, connectivity between all cluster nodes will be returned.
+func (c *Client) NetworkDiagnostics(ref string) (types.ConnectivityResults, error) {
```

Comment:
We should probably accept some sort of `context` to allow cancellation.

Reply:
agreed, had a quick look about adding it and realised I'd be creating yet another function signature.  After quick discussion with Dom we agreed we'd refactor the api to use contexts more consistently soon, but that we'd leave this as-is for now.